dev/TabView/TabView.cpp

```diff
@@ -0,0 +1,249 @@
+                int index = startIndex;
+
+                do
+                {
+                    auto nextItem = ContainerFromIndex(index).as<winrt::ListViewItem>();
+
+                    if (nextItem && nextItem.IsEnabled() && nextItem.Visibility() == winrt::Visibility::Visible)
+                    {
+                        // We need to wait until OnSelectionChanged fires to change the selection, otherwise it will get lost.
+                        m_indexToSelectOnSelectionChanged = std::optional(index);
+                        break;
+                    }
+
+                    // try the next item
+                    index++;
+                    if (index >= numItems)
+                    {
+                        index = 0;
+                    }
+                } while (index != startIndex);
+            }
+        }
+    }
+
+    UpdateTabWidths();
+
+    __super::OnItemsChanged(item);
+}
+
+void TabView::OnSelectionChanged(const winrt::IInspectable&, const winrt::SelectionChangedEventArgs&)
+{
+    if (m_indexToSelectOnSelectionChanged)
+    {
+        SelectedItem(Items().GetAt(m_indexToSelectOnSelectionChanged.value()));
+        m_indexToSelectOnSelectionChanged = {};
+    }
+
+    UpdateTabContent();
+}
+
+void TabView::UpdateTabContent()
+{
+    if (auto tabContentPresenter = m_tabContentPresenter.get())
+    {
+        if (!SelectedItem())
+        {
+            tabContentPresenter.Content(nullptr);
+            tabContentPresenter.ContentTemplate(nullptr);
+        }
+        else
+        {
+            if (auto container = ContainerFromItem(SelectedItem()).as<winrt::ListViewItem>())
+            {
+                tabContentPresenter.Content(container.Content());
+                tabContentPresenter.ContentTemplate(container.ContentTemplate());
+            }
+        }
+    }
+}
+
+void TabView::CloseTab(winrt::TabViewItem const& container)
+{
+    if (auto item = ItemFromContainer(container))
+    {
+        uint32_t index = 0;
+        if (Items().IndexOf(item, index))
+        {
+            auto args = winrt::make_self<TabViewTabClosingEventArgs>(item);
+
+            m_tabClosingEventSource(*this, *args);
+
+            if (!args->Cancel())
+            {
+                Items().RemoveAt(index);
+            }
+        }
+    }
+}
+
+void TabView::OnScrollDecreaseClick(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::max(0.0, scrollViewer.HorizontalOffset() - c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::OnScrollIncreaseClick(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::min(scrollViewer.ScrollableWidth(), scrollViewer.HorizontalOffset() + c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::UpdateTabWidths()
+{
+    double tabWidth = DoubleUtil::NaN;
```

Comment:
>DoubleUtil::NaN [](start = 22, length = 15)

please delete this helper -- let's use standard (documented) constants from STL:

```suggestion
    double tabWidth = std::numeric_limits<double>::quiet_NaN();
```

It looks like DoubleUtil::NaN is just an alias for this but the 10 characters saved of typing doesn't seem to warrant the extra complexity.